File: exercises/D802/solution.cpp
#include <iostream>
using namespace std;
int main() {
    int n, d, a[2000];
    cin >> n >> d;
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    int ans = 0;
    for (int i = 1; i < n; i++) {
      for (int j = 0; j < n - i; j++) {
        if (d == 0 && a[j] > a[j + 1] ||
            d == 1 && a[j] < a[j + 1]) {
          int t = a[j];
          a[j] = a[j + 1];
          a[j + 1] = t;
          ans++;
        }
      }
    }
    cout << ans << endl;
    for (int i = 0; i < n - 1; i++) {
      cout << a[i] << " ";
    }
    cout << a[n - 1] << endl;
}

File: exercises/D802/solution.py
n, d = map(int, input().split())
a = list(map(int, input().split()))
ans = 0
for i in range(1, n):
    for j in range(n - i):
        if d == 0 and a[j] > a[j + 1] or \
            d == 1 and a[j] < a[j + 1]:
            t = a[j]
            a[j] = a[j + 1]
            a[j + 1] = t
            ans += 1
print(ans)
for i in range(n - 1):
    print(a[i], end=' ')
print(a[n - 1])
